content/en/docs/concepts/workloads/pods/ephemeral-containers.md

```diff
@@ -0,0 +1,219 @@
+---
+reviewers:
+- verb
+- yujuhong
+title: Ephemeral Containers
+content_template: templates/concept
+weight: 80
+---
+
+{{% capture overview %}}
+
+{{< feature-state state="alpha" >}}
+
+This page provides an overview of ephemeral containers: a special type of container
+that runs temporarily in an existing pod to accomplish user-initiated actions such
+as troubleshooting. You use ephemeral containers to inspect services rather than
+build applications.
+
+{{< warning >}}
+Ephemeral containers are in early alpha state and are not suitable for production
+clusters. You should expect the feature not to work in some situations, like
+when targeting the namespaces of a container. In accordance with the [Kubernetes
+Deprecation Policy](/docs/reference/using-api/deprecation-policy/), this alpha
+feature could change significantly in the future or be removed entirely.
+{{< /warning >}}
+
+{{% /capture %}}
+
+{{% capture body %}}
+
+## Understanding ephemeral containers
+
+{{< glossary_tooltip text="Pods" term_id="pod" >}} are the fundamental building
+block of Kubernetes applications. Since pods are intended to be disposable and
+replaceable, you cannot add a container to a pod once it has been created.
+Instead, you usually delete and replace pods in a controlled fashion using
+[deployments](/docs/concepts/workloads/controllers/deployment/).
+
+Sometimes it's necessary to inspect the state of an existing pod, however, for
+example to troubleshoot a hard-to-reproduce bug. In these cases you can run
+an ephemeral container in an existing pod to inspect its state and run
+arbitrary commands.
+
+### Differences from regular containers
+
+Ephemeral containers lack guarantees for resources or execution, and they will
+never be automatically restarted, so they are not appropriate for building
+applications.  Ephemeral containers are described using the same `ContainerSpec`
+as regular containers, but many fields are incompatible and disallowed for
+ephemeral containers.
+
+- Ephemeral containers may not have ports, so fields such as `ports`,
+  `livenessProbe`, `readinessProbe` are disallowed.
+- Pod resource allocations are immutable, so setting `resources` is disallowed.
+- For a complete list of allowed fields, see the [EphemeralContainer reference
+  documentation](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#ephemeralcontainer-v1-core).
+
```

Comment:
nit: does this link to the reference docs exist? Ref docs for v1.16 have not been generated yet.